[msal-extensions] Change sample location. Add documentation and diagrams. (#2326)
AzureAD/microsoft-authentication-library-for-js: extensions/docs/msal-node-extensions.md

Supported platforms are Windows, Mac and Linux:

- Windows - DPAPI is used for encryption.
- MAC - The MAC KeyChain is used.
- Linux - LibSecret is used for storing to "Secret Service".

## Code
### Creating the persistence layer
Creating the persistence will differ based on what platform you are targeting.

#### Windows: 
```js
import { FilePersistenceWithDataProtection, DataProtectionScope } from "@azure/msal-node-extensions";

const cachePath = "path/to/cache/file.json"; 
const dataProtectionScope = DataProtectionScope.CurrentUser;
const optinalEntropy = "";
const windowsPersistence = FilePersistenceWithDataProtection.create(cachePath, dataProtectionScope, optionalEntropy);
```

- cachePath is the path in the file system where the encrypted cache file will be stored. 

> - dataProtectionScope specifies the scope fo the data protection - either the current user or the local machine. You do not need a key to protect or unprotect the data. If you set the scope to CurrentUser, only applications running on your credentials can unprotect the data; howerver, that mean that any application running on your credentials can access the protected data. If you set the scope to LocalMachine, any full-trust application on the computer can unprotect, access, and modify the data.

Reviewer: ```suggestion
- dataProtectionScope specifies the scope of the data protection - either the current user or the local machine. You do not need a key to protect or unprotect the data. If you set the scope to CurrentUser, only applications running on your credentials can unprotect the data; however, that means that any application running on your credentials can access the protected data. If you set the scope to LocalMachine, any full-trust application on the computer can unprotect, access, and modify the data.
```

> - optionalEntoropy specifies password or other additional entropy used to encrypt the data.

Reviewer: ```suggestion
- optionalEntropy specifies password or other additional entropy used to encrypt the data.
```

> The FilePersistenceWithDataProtection use the Win32 CryptProtectData and CryptUnprotectData APIs. For more information on dataProtectionScope, or optionalEntropy, reference the documentation for those APIs.  

Reviewer: ```suggestion
The FilePersistenceWithDataProtection uses the Win32 CryptProtectData and CryptUnprotectData APIs. For more information on dataProtectionScope, or optionalEntropy, reference the documentation for those APIs.  
```
